### PR TITLE
Switch media paths to dedicated domain (both new + legacy)

### PR DESCRIPTION
**Pull request by vinyll**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ deploy:
 	${remote} "${goto_src} && git fetch origin ${branch} && git checkout ${branch} && git reset --hard FETCH_HEAD"
 	${remote} "${goto_src} && pip install -r requirements.txt"
 	${remote} "${goto_src} && pkill gunicorn; \
-		gunicorn --daemon -b 127.0.0.1:5000 prod:app \
+		gunicorn --daemon -b 127.0.0.1:5000 preprod:app \
```
Comment (davidbgk) [inline Makefile:28]: 👍

File: config.py [hunk 2]
```diff
@@ -70,6 +72,10 @@ class BaseConfig:
     DEFAULT_LANGUAGE = LANGUAGES[0][0]
 
 
+class PreprodConfig(BaseConfig):
+    MEDIA_URL = 'https://preprod.cafebabel.com/media'
```
Comment (davidbgk) [inline config.py]: Switch to `https://media.preprod.cafebabel.com`

File: cafebabel/django_migrations.py [hunk 3]
```diff
@@ -151,7 +151,16 @@ def sanitize_title(title):
     return title
 
 
-def create_article(old_article):
+def sanitize_media_paths(app, content):
+    target = f'src="{app.config["MEDIA_URL"]}/archives'
+    return (content
+            .replace('src="http://m.cbabel.eu', target)
```
Comment (davidbgk) [inline cafebabel/django_migrations.py]: Remove the `/cache` prefix too.

File: cafebabel/django_migrations.py [hunk 4]
```diff
@@ -151,7 +151,16 @@ def sanitize_title(title):
...
+            .replace('src="http://m.cafebabel.com', target)
```
Comment (davidbgk) [inline cafebabel/django_migrations.py]: Same here.

File: cafebabel/django_migrations.py [hunk 5]
```diff
@@ -151,7 +151,16 @@ def sanitize_title(title):
...
+            .replace('src="/medias', target)
```
Comment (davidbgk) [inline cafebabel/django_migrations.py]: Same here?!

**Review by davidbgk — CHANGES_REQUESTED** [review]

**Comment by davidbgk** [comment]
Make sure that galleries images paths are fixed too!

File: cafebabel/core/mixins.py [hunk 6]
```diff
@@ -45,13 +45,14 @@ def delete_image(self):
     @classmethod
     def store_image(cls, sender, document, **kwargs):
         if document is not None and document._upload_image:
-            document.image_filename = secure_filename(
-                document._upload_image.filename)
+            image_filename = secure_filename(document._upload_image.filename)
+            document.image_filename = (f'/{document.upload_subpath}'
+                                       f'/{image_filename}')
```
Comment (vinyll) [inline cafebabel/core/mixins.py:50]: We're still having the risk of name conflict I guess if 2 pictures have the same name.
I [created an issue](https://github.com/cafebabel/cafebabel.com/issues/257) for this.

File: config.py [hunk 7]
```diff
@@ -63,7 +63,6 @@ class BaseConfig:
         'it': '@'.join(['redazione', 'cafebabel.com']),
     }
 
-    MEDIA_URL = 'https://media.cafebabel.com'
```
Comment (vinyll) [inline config.py]: How do you want to deal with the prod settings?
Should we explicitely override `MEDIA_URL = None` in some dev config?

**Review by vinyll — COMMENTED** [review]
r+wc 🚀

File: config.py (hunk 7 above)
Comment (davidbgk) [reply config.py]: I think we will have a dedicated (local) config + launcher for prod that's why I prefer to remove it from globals.

**Review by davidbgk — APPROVED** [review]

**Pull request merged**